test_app/tests/authentication/utils/test_user.py

```diff
@@ -0,0 +1,48 @@
+from unittest import mock
+
+import pytest
+
+from ansible_base.authentication.models import AuthenticatorUser
+from ansible_base.authentication.utils.user import can_user_change_password
+
+
+# Typeguard prevents None so we don't need to test that scenario
```

Comment:
I'm not fully to the point of trusting typeguard yet. Particularly because I don't understand how it resolves types of values coming from external code.

Like, will typeguard prevent this?

```python
# assume this library does NOT have type annotations
from cool_thirdparty_app.reallycool import a_cool_user

can_user_change_password(a_cool_user)
```

How can typeguard know what `a_cool_user` is in that case? What if `a_cool_user` is defined like this:

```python
a_cool_user = User.objects.first() if randint(1, 10) < 5 else None
```

I don't see how typeguard helps here, but maybe I'm just missing something.

Reply:
Because typeguard is enabled in tests we get this if we try to run with `(None, True)`:
```
FAILED test_app/tests/authentication/utils/test_user.py::test_can_user_change_password[None-True] - TypeError: argument of type 'NoneType' is not iterable
```

I changed the function to take `Optional(AbstractUser)` and the test is now valid.